coco_architecture.md

```diff
@@ -0,0 +1,147 @@
+  - Enclave (image management) downloads the container image(s) : **step 5 from the pic**
+  - Enclave verifies/decrypts the container image(s) : **step 6 from the pic**
+  - Enclave starts the container workload
+
+## Typical Confidential Container Stack using process based TEE (SGX)
+
+The following diagram shows how a CC solution is used to provide a confidential process to the end user: 
+
+![CC_SGX_container](https://github.com/confidential-containers/documentation/blob/main/images/CC_SGX_container.jpg)
+
+Following is the workflow when deploying a confidential container (Kubernetes POD). 
+We have marked in blue the changes from the confidential container using VM based TEE scenario: 
+
+- CC workload preparation
+  - User builds the container image(s)  (podman build -f …)
+  - User signs/encrypts the container image(s)
+  - User pushes the container image(s) to the image registry (podman push …)
+- Deploying the CC workload in k8s
+  - User deploys the workload (kubectl apply -f <...>)
+  - Kubernetes schedules the workload to target host having the required capability to run confidential containers
+- CC workload execution flow (**red connector** in the diagram)
+  - Confidential containers runtime on the host starts the enclave (SGX) 
+  - Enclave (agent) performs remote attestation: **steps 1-2 from pic**
+  - Enclave (agent) gets the keys required to verify/decrypt the containers image(s):  **steps 3-4 from the pic**
+  - Enclave (image management) downloads the container image(s) : **step 5 from the pic**
+  - Enclave verifies/decrypts the container image(s) : **step 6 from the pic**
+  - Enclave starts the container workload
+
+As can be seen, the flows for process and VM based TEEs are almost identical. It should be noted that process based TEE 
+requires a few additional software components like libOS without which the application requires re-architecting. 
+This is not the case for VM based TEEs.
+
+
+## Typical Confidential VM Stack
+
+The following diagram shows how a CC solution is used to provide a confidential VM to the end user: 
+
+![CC_TEE_VM](https://github.com/confidential-containers/documentation/blob/main/images/CC_TEE_VM.jpg)
+
+Following is the workflow when deploying a confidential VM:
+
+- CC workload preparation
+  - User builds the VM image(s) 
+  - User signs/encrypts the VM image(s)
+  - User pushes the VM image(s) to an image registry (not shown in the diagram) 
+- CC workload execution flow (**red connector** in the diagram)
+  - Confidential VM runtime on the host starts the VM TEE
+  - Enclave (agent) performs remote attestation: **steps 1-2 from pic**
+  - Enclave (agent) gets the keys:  **steps 3-4 from the pic**
+  - Enclave (agent) makes the keys available to the application: **step 5**
+  - The application will start running using the provided keys 
+
+
+# CNCF Confidential containers 
+
+## Overview 
```

Comment:
I don't understand the goal of this section. It seems to repeat the content from above but with maybe more detail. I think it would be more direct to perhaps just use these diagrams above and forego another section. Also it uses our off-hand v1 terminology which may not map to our actual release nomenclature.

Reply:
The goal of this section is to take the previous general explanation of how to support confidential containers with VM based TEE or process based TEE (and confidential VM stack) and to zoom into how we support it in the CoCo project using kata and enclave-CC.

Feel free to add content to help clarify things